### PR TITLE
[MRG] Enable multiple discrete targets per recording

### PR DESCRIPTION
**Pull request by gemeinl**
-base datasets now accept a list of strings as target name
-as a result base datasets and window datasets will return multiple values as y in `__getitem__`
-added tests to verify this
-added an example to demonstrate how to get from instantiation of a dataset class to pytorch dataloader

### CODE REVIEW

**Comment by gemeinl** [comment]
Open points:
 - One has to see whether this breaks skorch when using a multitarget dataset with EEGClassifier
 - It is unclear to me how this would work with an event dataset. There one has to pass the multiplte targets through mne objects like Annotation

**Comment by gemeinl** [comment]
This idea was also mentioned in https://github.com/braindecode/braindecode/issues/257#issue-926197775.

File: examples/plot_tuh_discrete_multitarget.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+)
+
+
+###############################################################################
+# Iterating through the DataLoader gives batch_X as tensor(4 x n_channels x
+# 1000), batch_y as [tensor([4 x age of subject]), tensor([4 x gender of
+# subject])], and batch_ind.
+for batch_X, batch_y, batch_ind in dl:
+    break
+batch_X, batch_y, batch_ind
```
Comment (robintibor) [inline examples/plot_tuh_discrete_multitarget.py:104]: Right now seems really hard to understand in examples for me. Would be nicer if batch_X, batch_y, batch_ind are shown separately in their own output cells

**Comment by robintibor** [comment]
Conflicts to be merged apparently @gemeinl .. see also comment above for examples

File: examples/plot_tuh_discrete_multitarget.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    preload=False,
+    add_physician_reports=False,
+)
+tuh.description
+
+
+###############################################################################
+# Iterating through the dataset gives x as ndarray(n_channels x 1) as well as
+# the target as [age of the subject, gender of the subject].
+x, y = tuh[0]
```
Comment (robintibor) [inline examples/plot_tuh_discrete_multitarget.py:47]:
```suggestion
x, y = tuh[-1]
```

File: examples/plot_tuh_discrete_multitarget.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+)
+# store the number of windows required for loading later on
+tuh_windows.set_description({
+    "n_windows": [len(d) for d in tuh_windows.datasets]})
+
+
+###############################################################################
+# Iterating through the dataset gives x as ndarray(n_channels x 1000), y as
+# [age, gender], and ind.
+x, y, ind = tuh_windows[0]
```
Comment (robintibor) [inline examples/plot_tuh_discrete_multitarget.py:75]:
```suggestion
x, y, ind = tuh_windows[-1]
```

File: examples/plot_tuh_discrete_multitarget.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    mapping={'M': 0, 'F': 1},  # map non-digit targets
+)
+# store the number of windows required for loading later on
+tuh_windows.set_description({
+    "n_windows": [len(d) for d in tuh_windows.datasets]})
+
+
+###############################################################################
+# Iterating through the dataset gives x as ndarray(n_channels x 1000), y as
+# [age, gender], and ind.
```
Comment (robintibor) [inline examples/plot_tuh_discrete_multitarget.py:74]:
```suggestion
# [age, gender], and ind. Let's look at the last example.
```

File: examples/plot_tuh_discrete_multitarget.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    target_name=('age', 'gender'),  # use both age and gender as decoding target
+    preload=False,
+    add_physician_reports=False,
+)
+tuh.description
+
+
+###############################################################################
+# Iterating through the dataset gives x as ndarray(n_channels x 1) as well as
+# the target as [age of the subject, gender of the subject].
```
Comment (robintibor) [inline examples/plot_tuh_discrete_multitarget.py:46]:
```suggestion
# the target as [age of the subject, gender of the subject]. Let's look at the last example
# as it has more interesting age/gender labels.
```

File: examples/plot_tuh_discrete_multitarget.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+dl = DataLoader(
+    dataset=tuh_windows,
+    batch_size=4,
+)
+
+
+###############################################################################
+# Iterating through the DataLoader gives batch_X as tensor(4 x n_channels x
+# 1000), batch_y as [tensor([4 x age of subject]), tensor([4 x gender of
+# subject])], and batch_ind.
```
Comment (robintibor) [inline examples/plot_tuh_discrete_multitarget.py:93]:
```suggestion
###############################################################################
# Iterating through the DataLoader gives batch_X as tensor(4 x n_channels x
# 1000), batch_y as [tensor([4 x age of subject]), tensor([4 x gender of
# subject])], and batch_ind. We will iterate to the end to look at the last example
# again.
```

File: examples/plot_tuh_discrete_multitarget.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+for batch_X, batch_y, batch_ind in dl:
+    break
```
Comment (robintibor) [inline examples/plot_tuh_discrete_multitarget.py:95]:
```suggestion
for batch_X, batch_y, batch_ind in dl:
    pass
```

**Comment by robintibor** [comment]
a-m-a-z-i-n-g

**Pull request merged**